### PR TITLE
Allow resizing of float boxes by edges

### PR DESCRIPTION
**Pull request by Triazic**
At present, float boxes are only resizable by the user via dragging the corners. This PR adds the ability to resize by dragging the edges, ala:
ResizeFloatsByEdge
:

I fully expect you to reject this PR as it does not even attempt to follow the existing code style etc. I only submit the PR for your reference. The code works fine and I am using the fork in my app.

### CODE REVIEW

**Comment by rinick** [comment]
Thank you for the contribution, 
I will implement that in the next version.

File: lib/DockPanel.js [hunk 1]
```diff
@@ -103,6 +105,54 @@ class DockPanel extends react_1.default.PureComponent {
...
+        this.onPanelEdgeStartDragB = (e) => this.onPanelEdgeDragStart(e, "b");
+        this.onPanelEdgeStartDragL = (e) => this.onPanelEdgeDragStart(e, "l");
+        this.onPanelEdgeStartDragT = (e) => this.onPanelEdgeDragStart(e, "t");
+        this.onPanelEdgeDragMove = (e) => {
+            let { panelData } = this.props;
+            if (e.screenX <= 0 || e.screenY <= 0)
+                return; // no idea why this happens, but eh
+            let dx = e.screenX - this._initX;
+            let dy = e.screenY - this._initY;
+            if (this._movingEdge == 't') {
```
Comment (bikkimahato) [inline lib/DockPanel.js:118]: Please use strict equality operator ===

Comment (Triazic) [reply lib/DockPanel.js:118]: I could, but my impression is that rinick will re-implement this anyway with their own code. 
Please advise if we will tweak and pull in this PR or have rinick code.

**Comment by rinick** [comment]
this is now implemented in 3.2.8

**Comment by Triazic** [comment]
I have installed version 3.2.8.
The feature does not (seem to) work. Attempting to drag edges has no effect.
image

**Comment by rinick** [comment]
@Triazic it works in the example pages, maybe you are using a customized css file and forget to update?

**Comment by Triazic** [comment]
> @Triazic it works in the example pages, maybe you are using a customized css file and forget to update?

ah yep, I am using customised css. for now I'm back to my own fork, will try porting to yours at some point

**Pull request closed without merging**